pkg/model/api/v1/secret/oauth.go

```diff
@@ -0,0 +1,154 @@
+package secret
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+type PublicOAuth struct {
+	httpClient       *http.Client
+	tlsConfig        *TLSConfig
+	ClientID         string           `json:"client_id" yaml:"client_id"`
+	ClientSecret     Hidden           `json:"client_secret" yaml:"client_secret"`
+	ClientSecretFile string           `json:"client_secret_file" yaml:"client_secret_file"`
+	TokenURL         string           `json:"token_url" yaml:"token_url"`
+	Scopes           []string         `json:"scopes" yaml:"scopes"`
+	EndpointParams   url.Values       `json:"endpoint_params" yaml:"endpoint_params"`
+	AuthStyle        oauth2.AuthStyle `json:"auth_style" yaml:"auth_style"`
+}
+
+func NewPublicOAuth(oauth *OAuth, tlsConfig *TLSConfig) *PublicOAuth {
+	if oauth == nil {
+		return nil
+	}
+
+	return &PublicOAuth{
+		tlsConfig:        tlsConfig,
+		ClientID:         oauth.ClientID,
+		ClientSecret:     Hidden(oauth.ClientSecret),
+		ClientSecretFile: oauth.ClientSecretFile,
+		TokenURL:         oauth.TokenURL,
+		Scopes:           oauth.Scopes,
+		EndpointParams:   oauth.EndpointParams,
+		AuthStyle:        oauth.AuthStyle,
+	}
+}
+
+type OAuth struct {
+	token            *oauth2.Token
+	tlsConfig        *TLSConfig
+	ClientID         string           `json:"client_id" yaml:"client_id"`
```

Comment:
when we are putting resources in pkg/model/api, we are enforcing the usage of the camelCase for YAML and JSON tags.

```suggestion
ClientID string `json:"clientID" yaml:"clientID"`
```